### PR TITLE
Conveniece method for namespacing

### PR DESCRIPTION
**Pull request by antstorm**
Hey,

Here's a bit of refactoring around the use of `NAMESPACE`. Removes some duplication and a lot of string concatenation.

### CODE REVIEW

File: src/remodal.js [hunk 1]
```diff
@@ -431,6 +422,20 @@
...
 
   /**
+   * Generates a string separated by dashes and prefixed with NAMESPACE
+   * @private
+   * @param str, str, ...
+   * @returns str
+   */
+  function namespaced() {
+    // Convert argument list into an array
+    var parts = Array.prototype.slice.call(arguments);
```
Comment (VodkaBears) [inline src/remodal.js]: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments

File: src/remodal.js [hunk 2]
```diff
@@ -431,6 +422,20 @@
   }
 
   /**
+   * Generates a string separated by dashes and prefixed with NAMESPACE
+   * @private
+   * @param str, str, ...
+   * @returns str
+   */
+  function namespaced() {
```
Comment (VodkaBears) [inline src/remodal.js]: So much costly actions to avoid concatenation :)

File: src/remodal.js [hunk 3]
```diff
@@ -274,37 +274,28 @@
    * @param {String} Reason of a state change.
    */
   function setState(instance, state, isSilent, reason) {
+
+    var newState = namespaced('is', state);
+    var allStates = [namespaced('is', STATES.CLOSING),
```
Comment (VodkaBears) [inline src/remodal.js]: This variables i missed.
+

File: src/remodal.js (hunk 2 above)
Comment (antstorm) [inline src/remodal.js]: Well Array creation is hardly a costly action and `join` is even faster than string concatenation on Firefox and Opera — http://jsperf.com/array-join-vs-string-connect

Comment (VodkaBears) [inline src/remodal.js]: A bit faster. And i guess not in this case.

**Comment by antstorm** [comment]
@VodkaBears I changed the `namespaced` method, taking your comments into account. Didn't know that `arguments` was leaking so badly. Hopefully the method is now much more lightweight.

File: src/remodal.js [hunk 4]
```diff
@@ -431,6 +422,22 @@
   }
 
   /**
+   * Generates a string separated by dashes and prefixed with NAMESPACE
+   * @private
+   * @param str, str, ...
```
Comment (VodkaBears) [inline src/remodal.js]: `@param {...String}`

Comment (antstorm) [inline src/remodal.js]: Sure, will change that. Are there any docs on how these comments should be generated?

File: src/remodal.js (hunk 2 above)
Comment (VodkaBears) [inline src/remodal.js]: Let's use a verb in the base form. Maybe `classify` or `namespacify`^^.

File: src/remodal.js [hunk 5]
```diff
@@ -431,6 +422,22 @@
...
+   * @returns str
+   */
+  function namespaced() {
+    var result = NAMESPACE;
+
+    for (var i = 0; i < arguments.length; ++i) {
```
Comment (VodkaBears) [inline src/remodal.js]: Not so critical in this case, but: `for (var i = 0, len = arguments.length; i < len; ++i)`

File: src/remodal.js [hunk 6]
```diff
@@ -431,6 +422,22 @@
   }
 
   /**
+   * Generates a string separated by dashes and prefixed with NAMESPACE
+   * @private
+   * @param {...String}
+   * @returns {String}
+   */
+  function namespaced() {
```
Comment (VodkaBears) [inline src/remodal.js]: Let's use a verb in the base form. Maybe `classify` or `namespacify`^^.

**Comment by VodkaBears** [comment]
Make changes, please, and i will merge it.
:8ball:

File: src/remodal.js (hunk 4 above)
Comment (VodkaBears) [inline src/remodal.js]: http://usejsdoc.org/tags-param.html#multiple-types-and-repeatable-parameters
https://developers.google.com/closure/compiler/docs/js-for-compiler?csw=1
http://stackoverflow.com/a/7954617

They are different everywhere. But i prefer to write a type with a capitalized first letter.

File: src/remodal.js (hunk 5 above)
Comment (antstorm) [inline src/remodal.js]: Well, `length` is an argument on `arguments` object, so calling it has **zero** extra cost and there's no need for caching it locally.

Comment (VodkaBears) [inline src/remodal.js]: kk

**Comment by VodkaBears** [comment]
Can you, please, make same PR to the master branch?

**Comment by antstorm** [comment]
Sure, no problem

**Pull request merged**